modules/openapi-generator/src/main/resources/php/HeaderSelector.mustache

```diff
@@ -72,8 +72,8 @@ class HeaderSelector
     {
         if (count($accept) === 0 || (count($accept) === 1 && $accept[0] === '')) {
             return null;
-        } elseif (preg_grep("/application\/json/i", $accept)) {
-            return 'application/json';
+        } elseif ($jsonAccept = preg_grep('~(?i)^(application/json|[^;/ \t]+/[^;/ \t]+[+]json)[ \t]*(;.*)?$~', $accept)) {
```

Comment:
```
(?i)^(application/json|[^;/ \t]+/[^;/ \t]+[+]json)[ \t]*(;.*)?$
```

That's the one I provided earlier. In the code above, it's prefix and suffix with `~`.

Does it work for you? Have it tested it locally?

Reply:
yes, see https://3v4l.org/TkPqE. In PHP that's the de delimiter (`~`). I used a diffferent delimiter then `/`, to avoid extra escaping.

Reply:
Thanks for the explanation.